### PR TITLE
UI | Implement simple database interactions

### PR DESCRIPTION
**Pull request by theusaf**
## Description

* Adds the missing `weight` field to the `Detection` model
* Loads detections from the database
* Adds some fake initial data at the start if not existing (debug mode)
* Adds some tests for methods for `Detection`
* Adds tests for how the titles of detections are formatted
  * This can be discussed in a future issue

## Related Issues
* References #8
* References #19

## Screenhots
<details>

image

image

</details>

### CODE REVIEW

File: lib/pub_sub/subscriber.dart
Comment (bpugh5) [inline lib/pub_sub/subscriber.dart:1]: Might be a little silly but could there be a comment at the top of this to explain what it does too?

Comment (bpugh5) [reply lib/pub_sub/subscriber.dart:1]: * for updatePreDetection specifically, not sure why I couldn't comment on one line

**Review by bpugh5 — APPROVED** [review]
Seems legit, thanks for throwing in the test data

**Review by bpugh5 — APPROVED** [review]
Updates look good, thanks for those comments.

**Review by kimberlymarkley — APPROVED** [review]

**Review by stalberm — APPROVED** [review]

**Pull request merged**